aodndata/acorn/handler.py

```diff
@@ -0,0 +1,115 @@
+        acorn_file_type = "radial_quality_controlled"
+    elif "FV01_wavespec.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-wave-spectra_QC"
+    elif "FV01_windp.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-wind-map_QC"
+    elif "FV01_wavep.nc" in filepath:
+        site_map_station = ('_CBG_', '_SAG_', '_ROT_', '_COF_')
+        if any(s in filepath for s in site_map_station):
+            acorn_file_type = "gridded_1h-avg-wave-site-map_QC"
+        else:
+            acorn_file_type = "gridded_1h-avg-wave-station-map_QC"
+    elif "FV00_1-hour-avg.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-current-map_non-QC"
+    elif "FV01_1-hour-avg.nc" in filepath:
+        acorn_file_type = "gridded_1h-avg-current-map_QC"
+    else:
+        raise InvalidFileNameError("file name: \"{filename}\" Unknown product type from filename".
+                                   format(filename=os.path.basename(filepath)))
+
+    return acorn_file_type
+
+
+class AcornHandler(HandlerBase):
+    def __init__(self, *args, **kwargs):
+        super(AcornHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = ['.nc']
+
+    def preprocess(self):
+        nc_collection = self.file_collection[0]
+
+        # enforce a check on the include_regexes and the filename to make sure the handler is processing the correct
+        # matching rexep file written in the config file
+        m = re.search(self.include_regexes[0], nc_collection._name)
+        if m is None:
+            raise InvalidFileNameError(
+                "file name: \"{filename}\" not allowed in this pipeline".format(
+                    filename=os.path.basename(nc_collection._name)))
+
+        # if file contains any of FILE_TYPE_NEED_INDEX, we index, otherwise publish only
+        if any(s in get_type(nc_collection._name) for s in FILE_TYPE_NEED_INDEX):
+            nc_collection.publish_type = PipelineFilePublishType.HARVEST_UPLOAD
+        else:
+            nc_collection.publish_type = PipelineFilePublishType.UPLOAD_ONLY
+
+        # check if file with same dest path already on s3. If yes, check it's date_created nc attribute to know
+        # if we need to overwrite this object or not
+        #     destination_s3 = self.dest_path(nc_collection._name)
+
+        #nc_collection.dest_path = os.path.join(destination_s3, nc_collection.name)
+        #storage_query_res = self.state_query.query_storage(destination_s3)
+        #if storage_query_res:
+        #    creation_date_nc_pipeline = self.get_creation_date(nc_collection)  # creation date of the file in the pipeline
+
+        #    #TODO get leigh to create the funcion below
+        #    nc_s3_path = download_s3(destination_s3)
+        #    creation_date_nc_s3 = get_creation_date(nc_s3_path)  # creation date of the file on s3
+        #
+        #    if creation_date_nc_pipeline > creation_date_nc_s3:
+        #        nc_collection.publish_type = PipelineFilePublishType.NO_ACTION
```

Comment:
Not to forget this section still needs to be written. Just waiting for @lwgordonimos to think about this functionality